cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -176,6 +179,41 @@ func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
 	return nil
 }
 
+func locationParameterFilterImpl(param azure.ArmTemplateParameterDefinition, location account.Location) bool {
+	if param.AllowedValues == nil {
+		return true
+	}
+
+	return slices.IndexFunc(*param.AllowedValues, func(v any) bool {
+		s, ok := v.(string)
+		return ok && location.Name == s
+	}) != -1
+}
+
+// defaultLocationToSelectFn resolves if there is an intention from a location parameter to use a default location.
+// If the parameter has the @allowedValues set, it will return the first value from the array as the default.
```

Comment:
Extra blank line here to make the comments pretty? 

```suggestion
// defaultLocationToSelectFn resolves if there is an intention from a location parameter to use a default location.
//
// If the parameter has the @allowedValues set, it will return the first value from the array as the default.
```